workspaces/util/src/__tests__/chunkBySize-test.ts

```diff
@@ -0,0 +1,78 @@
+import { describe, expect, it } from "@jest/globals";
+import { chunkBySize } from "../chunkBySize";
+
+describe("chunkBySize", () => {
+  it("splits an input array into smaller arrays", () => {
+    const result = chunkBySize([1, 2, 3, 4], 2);
+    expect([...result]).toStrictEqual([
+      [1, 2],
+      [3, 4],
+    ]);
+  });
+
+  it("handles an array not perfectly divisible by chunk size", () => {
+    const result = chunkBySize([1, 2, 3], 2);
+    expect([...result]).toStrictEqual([[1, 2], [3]]);
+  });
+
+  it("handles a chunk size greater than array length", () => {
+    const result = chunkBySize([1, 2], 3);
+    expect([...result]).toStrictEqual([[1, 2]]);
+  });
+
+  it("handles a chunk size of 1", () => {
+    const result = chunkBySize([1, 2, 3], 1);
+    expect([...result]).toStrictEqual([[1], [2], [3]]);
+  });
+
+  it("handles an empty array", () => {
+    const result = chunkBySize([], 2);
+    expect([...result]).toStrictEqual([]);
+  });
+
+  it("handles non-numeric elements", () => {
+    const result = chunkBySize(["a", "b", "c"], 2);
+    expect([...result]).toStrictEqual([["a", "b"], ["c"]]);
+  });
+
+  it("throws an error for a chunk size less than 1", () => {
+    expect(() => {
+      const result = chunkBySize([1, 2, 3], 0);
+      [...result];
+    }).toThrow("Chunk size must be a positive integer!");
+
+    expect(() => {
+      const result = chunkBySize([1, 2, 3], -1);
+      [...result];
+    }).toThrow("Chunk size must be a positive integer!");
+  });
+
+  it("throws an error for a non-integer chunk size", () => {
+    expect(() => {
+      const result = chunkBySize([1, 2, 3], 1.5);
+      [...result];
+    }).toThrow("Chunk size must be a positive integer!");
```

Comment:
I think the error type (RangeError) is more important than the error message. Unless the error message contains critical information, or we want to distinguish from some other error and we can't use the type, I usually don't think it's worth testing the error message itself.